src/IRCSock.cpp

```diff
@@ -1456,7 +1456,14 @@ void CIRCSock::ParseISupport(const CMessage& Message) {
             break;
         }
 
-        m_mISupport[sName] = sValue;
+        bool bAdding = '-' != sName[0];
```

Comment:
I think this introduces a vulnerability: try sending `:server 005 user =FOO :are supported`